taxcalc/functions.py

```diff
@@ -501,46 +516,31 @@ def StdDed(DSI, _earned, STD, p04470, e00100, e60000,
     if (MARS == 3 or MARS == 6) and (MIDR == 1):
         _standard = 0.
 
-    return (_standard, c04200, _numextra, c15200, c15100, x04500, _txpyers,
-            c04100)
+    return (_standard, c04200, _numextra, c15200, c15100, x04500, c04100)
 
 
 @iterate_jit(nopython=True, puf=False)
-def TaxInc(c00100, c04470, _standard, e37717, c21060, c21040,
-           p04470, c04200, c04500, c04600, x04500,
-           e04805, t04470, f6251, _exact, _feided, c04800, MARS,
+def TaxInc(c00100, _standard, e37717, c21060, c21040, c04500, c04600,
+           e04805, _feided, c04800, MARS,
            II_rt1, II_rt2, II_rt3, II_rt4,
            II_rt5, II_rt6, II_rt7, II_brk1, II_brk2, II_brk3,
-           II_brk4, II_brk5, II_brk6, _numextra, FDED, c02700):
-
-    # c04500 = c00100 - max(c04470, max(c04100, _standard + e37717))
+           II_brk4, II_brk5, II_brk6, c02700):
+    """
+    TaxInc function: ...
+    """
     c04500 = max(0, c00100 - max(c21060 - c21040, _standard + e37717))
 
-    # Check with Dan whether this is right!
-    # if c04470 > _standard:
```

Comment:
@martinholmer Sorry that the validation made you confused. 

First of all, most core team members do understand what's going on in the validation work, as I was tutored by them initially: the validation work has been performed prior to the time when I join the team. Also didn't mean to keep it secret or nontransparent, but there's nothing much to report if the empirical results were matched completely. 

When I mentioned SAS validation, it means that I'm comparing the **tax logic** of two calculators, and purely tax logic. I consider the extrapolation and imputation of Tax-Calculator, and the optimization of itemizing tax-logic-non-related. If we turn on the extrapolation and then comparing the two calculators, the work is meaningless, as we are feeding the two calculators (tax logic to be specific) with different inputs. To make sure we are really comparing tax logic of two calculators, there are a couple preliminaries to be done before the validation begins:

1. Extrapolation and imputation need to be switched off in Tax-calculator.
2. Optimization of itemization needs to be modified, as the code commented out here. 

Just like you mentioned, we need to feed the calculator with the **same** input, and if they yield the same output, the work is considered to be done. Without these two steps, the two calculator substantially won't have the same input. If we were not feeding the calculators with the same input, there's no way to perform the validation:

The first item listed above should be pretty straightforward that, after extrapolation and imputation, the real input for python calculator will be different from current `puf`, see #507 for details.

The second item might be a little twisted: optimizing in different ways will make some intermediate "input" to be different for two calculators. SAS calculator and python calculator are applying a different way on deciding whether a unit should itemize or not so that the unit could be better-off, which is considered an optimization. The way python is doing might be computationally expensive for SAS, since python tax logic is divided into many pieces of functions. PR #413 would be an example of how different ways of optimizing will yield various results. 

The two models, SAS calculator and python calculator to be specific, are somewhat different in terms of certain functionalities, and thus in general will not generate the same output even fed with the same input. What I'm trying to do here is to make sure two calculators to share the same functionalities, in order to verify and compare their tax logic, and purely tax logic. 

@MattHJensen @Amy-Xu Let me know if I missed something or said something wrong.